src/libraries/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs

```diff
@@ -82,24 +82,18 @@ public Matrix Clone()
             return new Matrix(clonedMatrix);
         }
 
-        public float[] Elements
+        public unsafe float[] Elements
```

Comment:
nit: There are users of this, like `OffsetX` and `OffsetY` where allocating the array is still "pointless". That is, it's an allocation that will live only until the next collection.

We should probably have:
```
internal void GetElements(Span<float> elements)
{
    Debug.Assert(elements.Length >= 6);

    fixed (float* m = elements)
    {
        Gdip.CheckStatus(Gdip.GdipGetMatrixElements(new HandleRef(this, NativeMatrix), m));
    }
}
```

We can then have `Elements`, `OffsetX`, and `OffsetY` all call into that; with `Elements` using an array and with `OffsetX`/`OffsetY` using a stack-allocated buffer.

We could probably also propose said API become public (with the appropriate error checking and `Try` equivalent).

Reply:
Yeah, I was going to follow up with that- I'll do that now.